File: leecode/code/287/287.java
class Solution {
    public int findDuplicate(int[] nums) {
        int tur = nums[nums[0]];
        int rabit = nums[nums[nums[0]]];


        while (rabit != tur) {
            tur = nums[tur];
            rabit = nums[nums[rabit]];
        }


        tur = nums[0];
        while (tur != rabit) {
            tur = nums[tur];
            rabit = nums[rabit];
        }
        
        return rabit;
    }
}

File: leecode/code/287/287.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow = 0
        fast = 0
        count = 1


        while fast != slow or count == 1:
            fast = nums[nums[fast]]
            slow = nums[slow]
            count += 1
        
        slow = 0
        while slow != fast:
            slow = nums[slow]
            fast = nums[fast]
        
        return fast